Source/Mosa.Korlib/System.Collections.Generic/List.cs

```diff
@@ -345,11 +345,11 @@ public void Insert(int index, T item)
 	{
 		EnsureCapacity(_size + 1);
 
-		_size++;
 		for (int i = index; i < _size; i++)
```

Comment:
This loop should start from the end and work backwards, otherwise it is overwriting as it moved forwards.

e.g.
[5, 6, 7]
Insert 9 at index 1
After first loop iteration: [5, 6, 6, 0]
After second loop iteration: [5, 6, 6, 6]
After final assignment at desired index: [5, 9, 6, 6]

Reply:
Thanks!